commands/dank_meme_poster.py

```diff
@@ -0,0 +1,188 @@
+from tempfile import NamedTemporaryFile
+from zipfile import ZipFile
+
+import discord
+import requests
+from PIL import Image
+from requests import HTTPError
+
+from commands.base import BaseCommand
+from discord_bot import MEDIA_PATH
+
+
+def _process_image(file):
+    """
+    processes image/gif and saves it to hard drive
+
+    Parameters
+    ----------
+    file : BytesIO
+
+    Returns
+    -------
+    None
+
+    """
+    file.seek(0)
+    image = Image.open(file)
+    if image.format == 'GIF':
+        image.save(os.path.join(MEDIA_PATH, '{}.gif'.format(len(os.listdir(MEDIA_PATH)))),
+                   save_all=True,
+                   optimize=True
+                   )
+    else:
+        image = image.convert(mode='RGB')
+        image.save(os.path.join(MEDIA_PATH, '{}.jpg'.format(len(os.listdir(MEDIA_PATH)))),
+                   format='JPEG',
+                   optimize=True
+                   )
+
+
+class DankMemeBulkUpload(BaseCommand):
+    trigger = 'dankbulkupload'
+    description = 'Uploads a ZIP of images to HuskieBot (via attachments or URL) for use with "!dank" command'
+
+    async def _download(self, url):
+        """
+        downlaods ZIP file from url
+
+        Parameters
+        ----------
+        url : str
+
+        Returns
+        -------
+        ZipFile
+
+        """
+        await self.client.wait_until_ready()
+        r = requests.get(url, stream=True)
+        if r.status_code == 200:
+            with NamedTemporaryFile() as temp:
+                for chunk in r.iter_content(chunk_size=4096):
+                    if chunk:  # filter out keep-alive new chunks
+                        temp.write(chunk)
+                    await asyncio.sleep(0.01)  # allows HuskieBot to respond to other requests
+                temp.flush()
+                temp.seek(0)
+                return ZipFile(temp.name, 'r')
+        else:
+            raise HTTPError('Received a non 200 status code: {}'.format(r.status_code))
+
+    async def command(self, message):
+        """
+        uploads a zip of images from attachments or url to HuskieBot's library
+
+        Parameters
+        ----------
+        message : discord.Message
+
+        Returns
+        -------
+        str
+
+        """
+        args = message.content.split(' ')[1:]
+        if message.attachments or len(args) == 1:
+            await message.channel.send('{} I am downloading the file. This may take a long time. '
+                                       'I will ping you when I finish.'
+                                       .format(message.author.mention))
+            file_count = 0
+            try:
+                zip_file = None
+                if message.attachments:
+                    for attachment in message.attachments:
+                        zip_file = await self._download(attachment.url)
+                        file_count += len(zip_file.infolist())
+                        for file in zip_file.infolist():
```

Comment:
Consider adding error handling for each file in the zip in case one has an issue, it will still process the rest